rholang/src/main/scala/coop/rchain/rholang/interpreter/PrettyPrinter.scala

```diff
@@ -174,16 +185,43 @@ case class PrettyPrinter(freeShift: Int, boundShift: Int) {
   private def buildPattern(patterns: Seq[Channel]): (Int, String) =
     ((0, "") /: patterns.zipWithIndex) {
       case ((patternsFree, string), (pattern, i)) =>
-        (patternsFree + freeCount(pattern), string + buildString(pattern) + {
-          if (i != patterns.length - 1) ", "
-          else ""
-        })
+        (patternsFree + freeCount(pattern),
+          string +
+            this
+              .copy(
+                freeId = boundId,
+                boundId = rotate(increment(baseId)),
+                baseId = increment(baseId)
+              )
+              .buildString(pattern) + {
+            if (i != patterns.length - 1) ", "
+            else ""
+          })
     }
 
   private def buildMatchCase(matchCase: MatchCase): String = {
     val patternFree: Int = matchCase.pattern.get.freeCount
-    PrettyPrinter(boundShift, 0).buildString(matchCase.pattern.get) + " => " +
-      PrettyPrinter(freeShift, boundShift + patternFree).buildString(matchCase.source.get)
+    "case " +
+      this
+        .copy(
+          freeShift = boundShift,
+          boundShift = 0,
+          freeId = boundId,
+          boundId = rotate(increment(boundId))
+        )
+        .buildString(matchCase.pattern.get) + " => " +
+      this
+        .copy(boundShift = boundShift + patternFree)
+        .buildString(matchCase.source.get)
+  }
+
+  private def rotate(id: String): String =
+    id.map(char => ((char + rotation - 97) % 26 + 97).toChar)
+
+  private def increment(id: String): String = {
+    val newId = ((id.last + 1 - 97) % 26 + 97).toChar
+    if (newId equals id(0)) id.dropRight(1) ++ newId.toString * 2
```

Comment:
This function does the wrong thing once you get to ba.
This is why it needed to be split.
The conditional should just be if newId equals 'a'

Reply:
Write some tests just for increment.